vividus-util/src/test/java/org/vividus/util/wait/DurationBasedWaiterTests.java

```diff
@@ -58,4 +58,19 @@ void shouldInvokeRunnableOnceBeforeReachingStopCondition() throws IOException
         new DurationBasedWaiter(Duration.ZERO, Duration.ZERO).wait(failableRunnable, Boolean.TRUE::booleanValue);
         verify(failableRunnable, times(1)).run();
     }
+
+    @Test
+    void shouldDecreasePollingTimeoutAccordinglyAfterLongRunningRunnable() throws IOException
+    {
+        FailableSupplier<Boolean, IOException> valueProvider = mock(FailableSupplier.class);
+        when(valueProvider.get())
+                .thenAnswer(invocation -> {
+                    Thread.sleep(5_000);
+                    return false;
+                })
+                .thenReturn(true)
+                .thenReturn(false);
+        assertTrue(new DurationBasedWaiter(new WaitMode(Duration.ofSeconds(10), 10))
```

Comment:
does it make sense to validate the quantity of provider calls here?

Reply:
no, the result check guarantees the provider is invoked properly, other edge cases are covered by another unit tests